src/cmap/wire_protocol/on_data.ts

```diff
@@ -116,6 +116,7 @@ export function onData(
     emitter.off('data', eventHandler);
     emitter.off('error', errorHandler);
     finished = true;
+    timeoutForSocketRead?.clear();
```

Comment:
This was the cause of the hanging test suites - some CSOT tests spawn huge timeouts that we never clean up.